integration-test/helpers/harness.js

```diff
@@ -59,21 +48,29 @@ const setup = async (ops) => {
     if (loadExtension) {
         // Grab a handle on the background page for the extension.
         try {
-            const backgroundPageTarget =
-                  await browser.waitForTarget(
-                      target => (
-                          target.type() === 'background_page' ||
-                          target.type() === 'service_worker'
-                      ),
-                      { timeout: 2000 }
-                  )
+            const targetFilter = target => (
+                target.type() === 'background_page' ||
+                target.type() === 'service_worker'
+            )
+            let backgroundPageTarget = await browser.targets().find(targetFilter)
```

Comment:
I don't think it's necessary to do `browser.targets().find(targetFilter)` first, we can just do `await browser.waitForTarget` initially, right?

Reply:
It's technically a race condition with their API I'm happy to remove but I don't think it harms being there either.

Reply:
I don't think it is necessary or a race condition. If the target already exists, I think `waitForTarget` should return it right away. See [the example in the Puppeteer docs](https://github.com/puppeteer/puppeteer/blob/main/docs/index.md#working-with-chrome-extensions) and [the docs for waitForTarget()](https://github.com/puppeteer/puppeteer/blob/main/docs/api/puppeteer.browser.waitfortarget.md).

Reply:
We're calling `waitForTarget` after the page has loaded, thus the service_worker or background_page might have been triggered in that time prior to us setting up the listener.
By definition it will race; unless they do something magic under the hood like pooling targets for a period of time for a listener to emerge... which seems doubtful.

I'm aware the docs don't have this but also they broke it themselves recently and didn't have tests for that.

Reply:
My understanding is that `waitForTarget` already checks for existing targets that match the predicate, before waiting for new ones to be created. See https://github.com/puppeteer/puppeteer/blob/bdcb748b98b621510de72f30b24f8c47fee0ad11/src/common/Browser.ts#L659

Reply:
Ah that seems non obvious :') thanks!